probe/http/http.go

```diff
@@ -188,7 +188,11 @@ func (h *HTTP) DoProbe() (bool, string) {
 	}
 	req.Header.Set("User-Agent", global.OrgProgVer)
 	for k, v := range h.Headers {
-		req.Header.Set(k, v)
+		if k == "Host" {
```

Comment:
In the HTTP protocol [1]:
>Each header field consists of a name followed by a colon (":") and the field value. Field names are case-insensitive.

It's better to judge it in a case-insensitive way.

[1]: https://www.rfc-editor.org/rfc/rfc9110.html#name-field-names